### PR TITLE
feat: add file.cp and file.mv functions

### LINKED ISSUE #1267: Extend file module functions (for smoother GitOps)
Hello there!

KCL being a very capable versatile language should have more file system control functions:
  - mkdir
  - delete
 And maybe even more than this.
 
It's essential to have control over transitional file creation/deletion and placement. For instance in my case, I do generate prometheus rules from parameterized KCL definitions as the first step:

```
cd appsops/rules/graph && mkdir -p rules manifests && kcl -dY ./kcl.yaml
generate_rules:
- graph/rules/common-certmanager.yaml
- graph/rules/common-elasticsearch.yaml
- graph/rules/common-istio.yaml
- graph/rules/common-k8s-events.yaml
- graph/rules/common-k8s-nodes.yaml
- graph/rules/common-k8s-resources.yaml
- graph/rules/common-vector.yaml
```

This is the configuration entry for the next stage, which proceeds with VMRule manifests generation. Simply there should be enough control in place on where to place the files, how to create or delete them to _enable smooth GitOps experience_ out of box. In fact KCL can be and should be used independently to provide a solid GitOps experience.

### PR DESCRIPTION
**Pull request by shruti2522**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 
fix #1267, provides additional file module functions

#### 2. What is the scope of this PR (e.g. component or file name):

kcl-lang/kcl/kclvm
#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

Added functions to copy and move files.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: kclvm/loader/src/snapshots/kclvm_loader__tests__builtin_call_1.snap [hunk 1]
```diff
@@ -383,6 +369,20 @@ expression: "format!(\"{:#?}\", p.symbols.values())"
                 },
                 kind: Value,
             },
+             SymbolRef {
```
Comment (Peefy) [inline kclvm/loader/src/snapshots/kclvm_loader__tests__builtin_call_1.snap:372]: The CI failed according to here.

Comment (shruti2522) [reply kclvm/loader/src/snapshots/kclvm_loader__tests__builtin_call_1.snap]: The error was arising due to the indentation of `SymbolRef`, fixed it.

**Review by Peefy — APPROVED** [review]
LGTM

**Review by zong-zhe — APPROVED** [review]
LGTM

**Pull request merged**